### ISSUE TITLE
Fix S3398 FP: Method used from multiple nested types

### ISSUE THREAD

**Issue by bart-vmware** [SonarSource/sonar-dotnet#8114]
### Description

When a private method is used from multiple nested types, the method cannot be moved as S3398 suggests.

### Repro steps

```c#
public class Outer
{
    private const int base = 42;

    private static void Print(int num)  // Noncompliant - static method is only used by the nested class, should be moved there
    {
        Console.WriteLine(num + base);
    }

    public class Nested1
    {
        public void SomeMethod()
        {
            Outer.Print(1);
        }
    }

    public class Nested2
    {
        public void SomeMethod()
        {
            Outer.Print(2);
        }
    }
}
```
### Expected behavior

No warning when the method is used from multiple nested types.

### Actual behavior

Warning that suggests a change that won't compile.

### Known workarounds

Turn off the rule entirely.

### Related information

* C# Plugins version: 9.11
* Visual Studio version: 17.7.4
* Operating System: Windows

**Comment by cristian-ambrosini-sonarsource** [comment]
Hi @bart-vmware!

Thank you for reaching out to us. Upon reviewing our test cases, it appears that a static method called by multiple sibling nested classes is already covered and should not trigger rule S3398.
To further validate this, I ran a test using the code snippet you provided and confirmed that no issue was raised on the 'Print' static method.

Could you please provide more details or a more comprehensive code sample to reproduce the FP?

**Comment by bart-vmware** [comment]
It does on my system, using the latest version. Repro steps:
1. File > New > Console App (target .NET 6)
1. From the package manager console: `Install-Package SonarAnalyzer.CSharp`
1. Copy/paste repro code, which does not compile because `base` is a reserved keyword, so escape using `@base` (I took this from https://sonarsource.github.io/rspec/#/rspec/S3398/csharp, so you may want to fix it there)
1. Build the solution

![image](https://github.com/SonarSource/sonar-dotnet/assets/104792814/f727dea5-942d-4b0f-a5b8-1c0d93537568)

**Comment by cristian-ambrosini-sonarsource** [comment]
> It does on my system, using the latest version. Repro steps:
> 
> 1. File > New > Console App (target .NET 6)
> 2. From the package manager console: `Install-Package SonarAnalyzer.CSharp`
> 3. Copy/paste repro code, which does not compile because `base` is a reserved keyword, so escape using `@base` (I took this from https://sonarsource.github.io/rspec/#/rspec/S3398/csharp, so you may want to fix it there)
> 4. Build the solution
> 
> ![image](https://user-images.githubusercontent.com/104792814/272214583-f727dea5-942d-4b0f-a5b8-1c0d93537568.png)

I can only see a warning for S3903, no green wavy lines or issues detected on the `Print` method. Am I missing something?
